Tools/ardupilotwaf/chibios.py

```diff
@@ -94,12 +95,20 @@ def wsl2_prereq_checks(self):
         try:
             where_python = subprocess.check_output('where.exe python.exe', shell=True, text=True)
         except subprocess.CalledProcessError:
-            #if where.exe can't find the file it returns a non-zero result which throws this exception
+            # if where.exe can't find the file it returns a non-zero result which throws this exception
             where_python = ""
-        if not where_python or not "\Python\Python" in where_python or "python.exe" not in where_python:
+        if not where_python or ("\Python\Python" not in where_python and "python.exe" not in where_python):
             print(self.get_full_wsl2_error_msg("Windows python.exe not found"))
             return False
-        return True
+        available_versions = where_python.split()
+        for test_version in available_versions:
+            drive_letter = test_version.split("\\")[0][0].lower()
+            corrected_path = "/" + os.path.join("mnt", drive_letter, *test_version.split("\\")[1:])
+            python_version = subprocess.check_output(corrected_path + " --version", shell=True, text=True)
```

Comment:
probably need a try/catch there:
```
/mnt/c/Users/PierreKancir/AppData/Local/Microsoft/WindowsApps/python.exe: Invalid argument
Waf: Leaving directory `/home/pierre/Workspace/ardupilot/build/CubeOrange'
Build failed
Traceback (most recent call last):
  File "/home/pierre/Workspace/ardupilot/modules/waf/waflib/Task.py", line 348, in process
    ret = self.run()
  File "/home/pierre/Workspace/ardupilot/Tools/ardupilotwaf/chibios.py", line 64, in run
    python_exe = self.wsl2_prereq_checks()
  File "/home/pierre/Workspace/ardupilot/Tools/ardupilotwaf/chibios.py", line 107, in wsl2_prereq_checks
    python_version = subprocess.check_output(corrected_path + " --version", shell=True, text=True)
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '/mnt/c/Users/PierreKancir/AppData/Local/Microsoft/WindowsApps/python.exe --version' returned non-zero exit status 1.
```
the path is correct but both python.exe and python3.exe exist but doesn't work. I got a working windows11 with wsl2
those may be left over from previous python installation IDK...

Reply:
so after some looking , we should get rid of AppData/Local/Microsoft/WindowsApps/python.exe that should open the windows store to propose to install python ...

Reply:
Added a filter that should ignore these python.exe paths.